ext/GLMSparseArraysExt.jl

```diff
@@ -1,94 +1,130 @@
 module GLMSparseArraysExt
 
 using GLM, LinearAlgebra, SparseArrays
-
+import GLM: AbstractWeights, UnitWeights, BlasReal, uweights
+using LinearAlgebra: LinearAlgebra
 ## QR
-mutable struct SparsePredQR{T,M<:SparseMatrixCSC,F} <: GLM.LinPred
+mutable struct SparsePredQR{T,M<:SparseMatrixCSC,F,W<:AbstractWeights} <: GLM.LinPred
     X::M                           # model matrix
     beta0::Vector{T}               # base vector for coefficients
     delbeta::Vector{T}             # coefficient increment
     scratchbeta::Vector{T}
     qr::F
+    wts::W
     scratch::M
 end
-function SparsePredQR(X::SparseMatrixCSC{T}) where {T}
+
+function SparsePredQR(X::SparseMatrixCSC{T}, wts::AbstractWeights) where {T}
     # The one(float(T))* part is because of a promotion issue in SPQR.jl on Julia 1.9
-    fqr = qr(sparse(one(float(T))*I, size(X)...))
-    return SparsePredQR{eltype(X),typeof(X),typeof(fqr)}(X,
-                                                         zeros(T, size(X, 2)),
-                                                         zeros(T, size(X, 2)),
-                                                         zeros(T, size(X, 2)),
-                                                         fqr,
-                                                         similar(X))
+    fqr = qr(sparse(one(float(T)) * I, size(X)...))
+    return SparsePredQR{eltype(X),typeof(X),typeof(fqr),typeof(wts)}(X,
+                                                                     zeros(T, size(X, 2)),
+                                                                     zeros(T, size(X, 2)),
+                                                                     zeros(T, size(X, 2)),
+                                                                     fqr,
+                                                                     wts,
+                                                                     similar(X))
 end
 
-GLM.qrpred(X::SparseMatrixCSC, pivot::Bool) = SparsePredQR(X)
+function GLM.qrpred(X::SparseMatrixCSC, pivot::Bool,
+                    wts::AbstractWeights=uweights(size(X, 1)))
+    SparsePredQR(X, wts)
+end
 
 function GLM.delbeta!(p::SparsePredQR{T}, r::Vector{T}, wt::Vector{T}) where {T}
     wtsqrt = sqrt.(wt)
     Wsqrt = Diagonal(wtsqrt)
     scr = mul!(p.scratch, Wsqrt, p.X)
     p.qr = qr(scr)
-    return p.delbeta = p.qr \ (Wsqrt*r)
+    p.delbeta = p.qr \ (Wsqrt * r)
```

Comment:
It's weird that this kind of change still happens. Did you use JuliaFormatter 2 (not 1)?